jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResponseUtils.java

```diff
@@ -1,86 +0,0 @@
-// ========================================================================
-//
-
-package org.eclipse.jetty.server;
-
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import org.eclipse.jetty.http.HttpField;
-import org.eclipse.jetty.http.HttpFields;
-import org.eclipse.jetty.http.HttpHeader;
-import org.eclipse.jetty.http.HttpHeaderValue;
-import org.eclipse.jetty.util.StringUtil;
-
-public class ResponseUtils
-{
-    public static void ensureConsumeAvailableOrNotPersistent(Request request, Response response)
-    {
-        if (request.consumeAvailable())
-            return;
-        ensureNotPersistent(request, response);
-    }
-
-    public static void ensureNotPersistent(Request request, Response response)
-    {
-        switch (request.getConnectionMetaData().getHttpVersion())
-        {
-            case HTTP_1_0 ->
-                // Remove any keep-alive value in Connection headers
-                response.getHeaders().computeField(HttpHeader.CONNECTION, (h, fields) ->
-                {
-                    if (fields == null || fields.isEmpty())
-                        return null;
-                    String v = fields.stream()
-                        .flatMap(field -> Stream.of(field.getValues()).filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s)))
-                        .collect(Collectors.joining(", "));
-                    if (StringUtil.isEmpty(v))
-                        return null;
-
-                    return new HttpField(HttpHeader.CONNECTION, v);
-                });
-
-            case HTTP_1_1 ->
-                // Add close value to Connection headers
-                response.getHeaders().computeField(HttpHeader.CONNECTION, (h, fields) ->
-                {
-                    if (fields == null || fields.isEmpty())
-                        return HttpFields.CONNECTION_CLOSE;
-
-                    if (fields.stream().anyMatch(f -> f.contains(HttpHeaderValue.CLOSE.asString())))
-                    {
-                        if (fields.size() == 1)
-                        {
-                            HttpField f = fields.get(0);
-                            if (HttpFields.CONNECTION_CLOSE.equals(f))
-                                return f;
-                        }
-
-                        return new HttpField(HttpHeader.CONNECTION, fields.stream()
-                            .flatMap(field -> Stream.of(field.getValues()).filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s)))
-                            .collect(Collectors.joining(", ")));
-                    }
-
-                    return new HttpField(HttpHeader.CONNECTION,
-                        Stream.concat(fields.stream()
-                                    .flatMap(field -> Stream.of(field.getValues()).filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s))),
-                                Stream.of(HttpHeaderValue.CLOSE.asString()))
-                            .collect(Collectors.joining(", ")));
-                });
```

Comment:
I think this is good... but I need to ponder for a bit.    I believe this logic is actually all contained within the HttpGenerator,  so that removing this method gets rid of duplication... but I need to convince myself that it is all there.  I'll take another day to walk through all the variations